lib/maths/COutliers.cc

```diff
@@ -377,8 +379,10 @@ CEnsemble<POINT>::computeOutlierScores(const std::vector<POINT>& points) const {
     TScorerVec scores(points.size());
     m_RecordMemoryUsage(core::CMemory::dynamicSize(scores));
 
+    std::size_t step{0ul};
     for (const auto& model : m_Models) {
         model.addOutlierScores(points, scores, m_RecordMemoryUsage);
+        state.nextStep(step++);
```

Comment:
IMHO, this constitutes a single step of the outlier detection job. WDYT @tveasey ?

Reply:
I agree. This seems like a natural place to me as well.